### DISCUSSION TITLE
Question about ZFS on Proxmox-Port

### DISCUSSION THREAD

**Discussion by rdeangel** [jiangcuo/Proxmox-Port#48] [Q&A]
First of all, thanks for the work you have put in this project, great work!

So the question:
I have proxmox installed on an Rpi4 8Gb and I have a 256Gb SSD card.
I want to have the ability to create snapshots for containers and it seems this only works if I create a virtual disk on ZFS or LVM.

I resized the proxmox partition to 30Gb and wanted to create a ZFS Disk on the remaining 220Gb so I can put containers that I can snapshot on there but I get the following error when I go to zfs:
"command '/sbin/zpool list -HpPLo name,size,alloc,free,frag,dedup,health' failed: exit code 1 (500)"

Am I doing something wrong? if so is there a way I can fix this?

Thanks for your help.

**Comment by rdeangel** [comment]
I resized the partition of Proxmox to 30Gb and created an LVM so that I can now use snapshots on the SD card but the question of ZFS still stands for things like HA failover with replication of VM disk to another pi.

If anyone cares here are the steps to create a local-lvm storage device on the same SD as where proxmox is installed. 
This is effectively exactly what you would get by default on a regular iso install of proxmox with a single disk on amd64.

----------------------------------------------------
An article on how to handle LVMs (https://medium.com/@yhakimi/lvm-how-to-create-and-extend-a-logical-volume-in-linux-9744f27eacfe)

resize partition on the sd card with gparted in another device (make sure it's not mounted)

then boot Proxmox and run the following commands (the below assumes you are logged in as root):

# create physical volume
the next partition should be /dev/mmcblk0p3, check with "fdisk -l"
```
pvcreate /dev/mmcblk0p3
```

# check you've created the physical volume
```
pvdisplay
```

# create volume group
```
vgcreate pve /dev/mmcblk0p3
```

# check the vg name pve shows under the physical volume
```
pvdisplay
vgs
```

# create logical volume
```
lvcreate --name data -l 99%FREE pve
```

# then convert the logical volume into a thin-volume as described in this page https://pve.proxmox.com/wiki/Storage:_LVM_Thin
```
lvconvert --type thin-pool pve/data
```

# then just add the following line at the bottom in this file "/etc/pve/storage.cfg"
```
lvmthin: local-lvm
        thinpool data
        vgname pve
        content rootdir,images
```


# after you've added the above /etc/pve/storage.cfg should look like this:
```
dir: local
        path /var/lib/vz
        content snippets,iso,vztmpl,backup,images,rootdir
        prune-backups keep-all=1

lvmthin: local-lvm
        thinpool data
        vgname pve
        content rootdir,images
```

**Comment by jiangcuo** [comment]
zfs:   https://github.com/jiangcuo/Proxmox-Arm64/issues/45

**Reply by rdeangel** [reply]
Thanks I'll try that when I get a chance.

**Reply by rdeangel** [reply]
that worked but I ran into some other issues so I just reinstalled it on rpi4 with EUFI boot on SD card and install proxmox on external USB SSD.
